Python/compile.c

```diff
@@ -8533,9 +8538,7 @@ remove_redundant_jumps(cfg_builder *g) {
             }
         }
     }
-    if (removed) {
-        eliminate_empty_basic_blocks(g);
-    }
+    assert(no_empty_basic_blocks(g));
```

Comment:
Even if this function cannot create empty basic blocks, an assert at the end makes no sense, as the input CFG might have empty blocks. 

Remind me, does a block with just NOPs count as empty, or is that just a block with `b_iused == 0`?

Reply:
> Even if this function cannot create empty basic blocks, an assert at the end makes no sense, as the input CFG might have empty blocks.

Right, following @sweeneyde 's review I added it as a precondition in this function, so I can remove it as a postcondition.

> 
> Remind me, does a block with just NOPs count as empty, or is that just a block with `b_iused == 0`?

Just  `b_iused == 0`.
(NOPs can make it to the final bytecode for line numbers).